File: 1161.maximum-level-sum-of-a-binary-tree/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func maxLevelSum(root *TreeNode) int {
    maxSum := math.MinInt
    maxLevel := 0
    level := 0
    q := []*TreeNode{root}
    for len(q) > 0 {
        level++
        levelSum := 0
        for i := len(q); i > 0; i-- {
            node := q[0]
            q = q[1:]
            levelSum += node.Val
            if node.Left != nil {
                q = append(q, node.Left)
            }
            if node.Right != nil {
                q = append(q, node.Right)
            }
        }
        if levelSum > maxSum {
            maxSum = levelSum
            maxLevel = level
        }
    }
    return maxLevel
}

File: 1161.maximum-level-sum-of-a-binary-tree/solution.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun maxLevelSum(root: TreeNode?): Int {
        var maxSum = Int.MIN_VALUE
        var maxLevel = 0
        var currentLevel = 1
        var nodes = listOfNotNull(root)

        while (nodes.isNotEmpty()) {
            val currentSum = nodes.sumOf { it.`val` }
            if (currentSum > maxSum) {
                maxSum = currentSum
                maxLevel = currentLevel
            }
            currentLevel++
            nodes = nodes.flatMap { listOfNotNull(it.left, it.right) }
        }
        return maxLevel
    }
}